pkg/manifest/manifest.go

```diff
@@ -0,0 +1,108 @@
+package manifest
+
+import (
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/ddev/ddev/pkg/globalconfig"
+	"github.com/ddev/ddev/pkg/manifest/storages"
+	"github.com/ddev/ddev/pkg/manifest/types"
+	"github.com/ddev/ddev/pkg/util"
+)
+
+func NewManifest(updateInterval time.Duration) *Manifest {
+	manifest := &Manifest{
+		fileStorage: storages.NewFileStorage(getLocalFileName()),
+		// TODO change to ddev repo before merge
+		githubStorage:  storages.NewGithubStorage(`gilbertsoft`, `ddev`, `manifest.json`, storages.Options{Ref: "task/upstream-infos"}),
+		updateInterval: updateInterval,
+	}
+	manifest.loadFromLocalStorage()
+
+	return manifest
+}
+
+const localFileName = ".manifest"
+
+type Manifest struct {
+	Manifest types.Manifest
+
+	fileStorage   types.ManifestStorage
+	githubStorage types.ManifestStorage
+
+	updateInterval time.Duration
+
+	mu sync.RWMutex
+}
+
+func (m *Manifest) Write() {
+	runTime := util.TimeTrack(time.Now(), "Write()")
+	defer runTime()
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	err := m.fileStorage.Push(m.Manifest)
+
+	if err != nil {
+		util.Error("Error while writing manifest: %s", err)
+	}
+}
+
+func (m *Manifest) loadFromLocalStorage() {
+	runTime := util.TimeTrack(time.Now(), "loadFromLocalStorage()")
+	defer runTime()
+
+	m.mu.Lock()
+	defer func() {
+		m.mu.Unlock()
+		/*go*/ m.updateFromGithub()
```

Comment:
What does the /*go*/ notation imply?

Reply:
go starts the method asynchronously, non blocking though. This has the effect the update will be done in the background withouth having the user to wait for. Commented here for testing purposes, is now active again.